### DISCUSSION TITLE
Has anyone integrated Cypress with Locust?

### DISCUSSION THREAD

**Discussion by cjcrandall** [cypress-io/cypress#16562] [Questions and Help]
Locust is great for performance testing API endpoints, but we'd also like to test our UI performance (e.g. page load times). I've seen an [example](https://github.com/SvenskaSpel/locust-plugins/blob/master/locust_plugins/users/webdriver.py) for integrating with webdriver but I'm curious if it'd be possible to create a Locust plugin (or in the other direction, a Cypress plugin) to integrate Locust and Cypress, or if anyone has already done something.

**Comment by sandra-ouadghiri** [comment]
Hi,
Not really a reply but
- Performance framework are coded with in mind the fact that the framework should not impact the measured metric. Cypress is a functional framework and thus probably doesn't follow this rule.
- For page load time, you can check how long takes a GET call to a given page, or you can use tools dedicated to web front performances such as lighthouse (feeded with urls)
- If you use a 'client journey' scenario you have in Cypress and plug that to a perf framework, what will it teach you? Time could remain the same but some steps could take way longer than other. If time is overall longer, you will need to dig into the failure. So for a 'live' performance tests where you look at your monitoring, why not, but for a pipeline usage it seems like a bad idea.

I would rather advice to have short very precise perf scenarios on given API calls (including loading the page) or UI stats (feeding the tool with URL and eventual headers to access pages behind authentication or other prerequisites) than trying to mix functional & performances.

This said, maybe you can achieve what you asked by having the spec code as a library shared between Cypress & Locust?
If someone has contradictory opinions, or knows about eventual exisiting plugins matching your question, hopefully they'll share; I'm curious so don't hesitate :) !!!
